jax/interpreters/partial_eval.py

```diff
@@ -742,8 +742,8 @@ def getconstvar(c):
       consts[v] = recipe.val
     elif isinstance(recipe, Literal):
       t_to_var[id(t)] = recipe
-    elif recipe is core.unit:
-      t_to_var[id(t)] = core.unitvar
+    elif recipe is None:
+      assert False
```

Comment:
This case arose when we had a Tracer created via `JaxprTracer.new_const` but not "instantiated", i.e. not corresponding to an "unknown" value in the jaxpr being built, and that Tracer was then an output of the traced function. Hence why, before removing units, it would always correspond to a unit-valued output when the jaxpr was built. But given the remove-units changes from last week, we no longer have such placeholders for known value outputs.